File: Demos/Cordova/CordovaPluginDeviceDemo/CordovaPluginDeviceDemo/ViewController.m
// Copyright (c) 2014 Intel Corporation. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#import "ViewController.h"
#import <WebKit/WebKit.h>

@import XWalkView;

@interface ViewController () <WKNavigationDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    WKUserContentController* userContentController = [[WKUserContentController alloc] init];
    WKWebViewConfiguration* config = [[WKWebViewConfiguration alloc] init];
    config.userContentController = userContentController;
    WKWebView* webview = [[WKWebView alloc] initWithFrame:self.view.frame configuration:config];

    webview.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    webview.frame = self.view.frame;
    webview.navigationDelegate = self;
    [self.view addSubview:webview];

    if ([[NSBundle mainBundle] objectForInfoDictionaryKey:@"CordovaPlugins"]) {
        NSString* namespace = @"xwalk.cordova";
        id ext = [XWalkExtensionFactory createExtension:namespace];
        [webview loadExtension:ext namespace:namespace thread:nil];
    }
    NSString* path = [[NSBundle mainBundle] pathForResource:@"index" ofType:@"html" inDirectory:@"www"];
    if (path.length) {
        [webview loadRequest:[NSURLRequest requestWithURL:[NSURL fileURLWithPath:path]]];
    }
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: src/cordova-plugin-demos/CordovaPluginFileDemo/CordovaPluginFileDemo/ViewController.swift
// Copyright (c) 2014 Intel Corporation. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import Foundation
import UIKit
import WebKit
import CrosswalkLite
import Cordova

class ViewController: CDVViewController, WKNavigationDelegate {

    override func viewDidLoad() {
        super.viewDidLoad()

        var userContentController = WKUserContentController()

        var config : WKWebViewConfiguration = WKWebViewConfiguration()
        config.userContentController = userContentController
        var webview = WKWebView(frame: view.frame, configuration: config)
        webview.autoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight
        webview.frame = view.frame
        webview.navigationDelegate = self
        view.addSubview(webview)

        if NSBundle.mainBundle().objectForInfoDictionaryKey("CordovaPlugins") != nil {
            webview.loadExtension("xwalk.cordova", namespace: nil, parameter: nil)
        }

        if let path = NSBundle.mainBundle().pathForResource("index", ofType: "html", inDirectory:"www") {
            webview.loadRequest(NSURLRequest(URL: NSURL.fileURLWithPath(path)!));
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }


}

